server/api.go

```diff
@@ -510,3 +531,25 @@ func LogMiddleware(typ string, opts LogOptions) func(next http.Handler) http.Han
 		})
 	}
 }
+
+func (info *SessionInfo) UnmarshalJSON(data []byte) error {
+	// Marshal in alias first to prevent infinite recursion
+	type Alias SessionInfo
+	err := json.Unmarshal(data, &struct{ *Alias }{(*Alias)(info)})
+	if err == nil && info.LDContext == LDContextSessionInfo {
```

Comment:
If `err != nil` then it should be logged, otherwise bugs resulting in non-nil errors will be hard to solve.